pinot-server/src/main/java/org/apache/pinot/server/api/resources/TableSizeResource.java

```diff
@@ -119,4 +122,33 @@ public String getTableSizeOld(
       throws WebApplicationException {
     return this.getTableSize(tableName, detailed);
   }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/tables/{tableName}/avgMemoryConsumedRealtime")
```

Comment:
Does it make sense to add it to `MmapDebugResource`?

Reply:
I don't have a strong preference. I put it here since this is at the table level whereas MmapDebugResource seems to be at the Server level. I'm fine either ways.

Reply:
If this is for debugging it should go into MmapDebugResource. If it is for building (say) a monitoring system, then it should be where you have put it.  On the other hand, for a monitoring system, you probably want to get the _current_ memory used, for consuming/completed etc. segments, for one or all tables. Does that make sense?

Also, you mentioned that it will be useful for subsequent provisioning? I don't understand that part. If we are provisioning for the same table, then yes, it will be useful, but the table is already provisioned. If we are provisioning for another table that looks and feels like the same table (i.e. data characteristics are very similar or same), then it will be useful as well. Even in these cases, the provisioning story will be different depending on ingestion rate of the new table.

But for any other table with completely different characteristics, this data will be useless.

Reply:
It is meant more for a continuous monitoring system. 

Good point about current memory: It probably makes more sense to get the latest size of the consuming segment for that table. Lemme add that as an additional endpoint.

Reply:
In that case, do the existing metrics help? I am sure you already gave it a thought, but I am curious to know what is lacking.